packages/flutter/lib/src/widgets/_window_win32.dart

```diff
@@ -680,6 +690,246 @@ class DialogWindowControllerWin32 extends DialogWindowController {
   }
 }
 
+/// Implementation of [TooltipWindowController] for the Windows platform.
+///
+/// {@macro flutter.widgets.windowing.experimental}
+///
+/// See also:
+///
+///  * [TooltipWindowController], the base class for tooltip windows.
+class TooltipWindowControllerWin32 extends TooltipWindowController
+    implements _WindowsMessageHandler {
+  /// Creates a new tooltip window controller for Win32.
+  ///
+  /// When this constructor completes the native window has been created and
+  /// has a view associated with it.
+  ///
+  /// {@macro flutter.widgets.windowing.experimental}
+  ///
+  /// See also:
+  /// * [TooltipWindowController], the base class for tooltip windows.
+  @internal
+  TooltipWindowControllerWin32({
+    required WindowingOwnerWin32 owner,
+    required TooltipWindowControllerDelegate delegate,
+    required BoxConstraints contentSizeConstraints,
+    required BaseWindowController parent,
+    required Rect anchorRect,
+    required WindowPositioner positioner,
+  }) : _delegate = delegate,
+       _owner = owner,
+       _parent = parent,
+       _anchorRect = anchorRect,
+       _positioner = positioner,
+       super.empty() {
+    _owner._addMessageHandler(this);
+    _onGetWindowPosition =
+        ffi.NativeCallable<
+          ffi.Pointer<_Rect> Function(
+            ffi.Pointer<_Size> childSize,
+            ffi.Pointer<_Rect> parentRect,
+            ffi.Pointer<_Rect> outputRect,
+          )
+        >.isolateLocal(_handleOnGetWindowPosition);
```

Comment:
This is the biggest type parameter I've ever seen. Any way to shorten it with a typedef or something? No worries if not, it just tripped me up.